V1.1 updates (#8)
PediatricOpenTargets/documentation: README.md

# Molecular Targets Platform Documentation 

The [Open Pediatric Cancer (OpenPedCan)](https://github.com/PediatricOpenTargets/OpenPedCan-analysis) project at the 
Children’s Hospital of Philadelphia, in partnership with the National Cancer Institute, is combining and harmonizing pediatric cancer datasets and integrating them into the Molecular Targets Platform <https://moleculartargets.ccdi.cancer.gov/> in order to accelerate pediatric cancer target identification and drug development. This is high-level overview of the Molecular Targets Platform data processing and analysis. For more information on the Molecular Targets Platform itself, see <https://moleculartargets.ccdi.cancer.gov/about>. 

---

> **Please note** that OpenPedCan is in continuous development and the GitHub repository main branch contents may not be identical to the Molecular Targets Platform site contents. While CHOP P30 DNA panel data is available through OpenPedCan, it is not available on the MTP site for this release.

Author: I argued against including language about the DGD DNA panels on Slack, but I guess FNL came back again and wanted something written down, which is why I included that line

> **Please note** that OpenPedCan is in continuous development and the GitHub repository main branch contents may not be identical to the Molecular Targets Platform site contents. While CHOP P30 DNA panel data is available through OpenPedCan, it is not available on the MTP site for this release.

Reviewer: haha ok

---

> For documentation on previous versions of the Molecular Targets Platform, please see [v1](past_versions/v1.md).

Author: Yes, agreed this will be unsustainable in the long term. I'll put it on the list to think about for the next update

## Datasets

While adult pan-cancer repositories have existed and accelerated cancer research for a decade, pediatrics cancers have been excluded, despite having different genetic and molecular etiologies than adult cancers. 
Over the past few years larger pediatric consortia, both disease-specific and pan-cancer, have tried to address this disparity. 
OpenPedCan is harmonizing the data from across these different consortia and presenting summarized data in one unified location in the pediatric MTP where it can be queried for associations between putative targets and pediatric cancers. 
As the project is ongoing, more data will continue to be added, but this current release includes four pediatric consortia datasets as well as GTEx data for comparisons to normal tissue expression:

![Image of a table giving the count of biospecimens for each sequencing type (DNA or RNA) for each tumor stage (primary or relapse) for each Dataset](figures/biospecimen_dataset_counts.png)

Primary tumors follow the standard definition of the original tumor in the body. 
However, for analysis purposes, MTP combines progressive, progressive post-mortem, and relapse tumors into our relapse tumor stage.

> If needed, more detailed sample information is available in the histology data provided through the [OpenPedCan-analysis GitHub repository](https://github.com/PediatricOpenTargets/OpenPedCan-analysis#how-to-obtain-openpedcan-data).

Author: Added a relapse description as mentioned in https://github.com/PediatricOpenTargets/documentation/pull/8#pullrequestreview-1200299378. How does this sounds?

Reviewer: perfect!

Once in the Cavatica workflow page, please click on the "Read All" link to open up the full documentation.

#### Somatic CNV Calling

Copy number variants are called using multiple tools: ControlFreeC, CNVkit, and GATK CNV. 
CNVkit calls are adjusted for sex using a pre-calculated germline sex estimate. 
Manta is used to call structural variants (SVs). 
All calls are made using GRCh38 references and variants are then annotated with gnomAD frequencies.
Publicly available files are further subjected to “germline masking,” which removes low frequency variants that could be used to identify the sample donor. 
For more details see the GitHub release at [Kid's First Somatic Variant Workflow](https://github.com/kids-first/kf-somatic-workflow) or to run the pipeline see the [CAVATICA App](https://cavatica.sbgenomics.com/public/apps/cavatica/apps-publisher/kfdrc-somatic-variant-workflow). 
Once in the Cavatica workflow page, please click on the "Read All" link to open up the full documentation. 

> Panel data was not reprocessed and the variant files provided by the submitters were merged with our harmonized data files.

Reviewer: ```suggestion
TARGET DNA Targeted Panel data was reharmonized in the same way as WGS and WXS descrribed herein.
CHOP P30 DNA Targeted Panel data was generated by the CHOP's DGD in hg19 and was lifted over to hg38.
```
I think it was something like this- we can ask in slack tomorrow.

### Somatic Alteration Data

#### Small Variants

> Multiple callers were used to determine single nucleotide variants (SNVs) since the literature suggests this reduces false positives. Using custom R scripts, a consensus SNV file was constructed, consisting only of SNVs that were called by 2 or more variant callers: GATK Mutect2, Strelka2, Lancet, and VarDict Java. 

Reviewer: ```suggestion
Multiple callers were used to determine single nucleotide variants (SNVs) since the literature suggests this reduces false positives. 
A consensus SNV file consisting only of SNVs that were called by 2 or more variant callers: GATK Mutect2, Strelka2, Lancet, and VarDict Java was created.
[Hotspot mutations](https://www.cancerhotspots.org/#/home) were annotated and recovered if called by only 1/4 algorithms above.
```
This was done in OpenPBTA within R, but now in cavatica so removed that part.

Annotations, including alternative gene and protein IDs and cancer references, were also added, see [the annotation calling workflow](https://github.com/kids-first/kf-somatic-workflow/blob/master/docs/kfdrc_annotation_subworkflow.md) for more details.

A unique variant id consisting of the hg38 coordinates and the reference and alternative alleles was created for consistency. 
Then several variant frequencies were calculated for each of those IDs within each cancer group and cohort. 
The frequency in the overall dataset, for each unique variant and gene, is the percentage of patients that have that variant or gene in the given cohort out of all patients in that cohort. 
The frequency in primary or relapse tumors, for each unique variant and gene, is the percentage of samples that have that variant or gene in the given cohort out of all samples in that cohort. 
Note that the frequencies and counts may not tally as expected for several reasons. 
First, the total columns use unique patients, while the primary/relapse tumor columns use unique samples. 

> Second, some submitters did not include information about the primary/relapse status of the samples, so those samples are omitted from the primary/relapse counts. 

Reviewer: this is not needed now, but may be easier for a user to see equations here

For more details on sample identity confirmation please see the [Kid's First NGS Checkmate Workflow](https://github.com/kids-first/ngs_checkmate_wf). 
Variants are called using GATK4 HaplotypeCaller. 
For more details on the alignment or to run the CAVATICA app yourself, see the GitHub release at [Kid's First Alignment and Haplotype Calling Workflow](https://github.com/kids-first/kf-alignment-workflow) and the [CAVATICA App](https://cavatica.sbgenomics.com/public/apps/cavatica/apps-publisher/kfdrc-alignment-workflow). 
Once in the Cavatica workflow page, please click on the "Read All" link to open up the full documentation.

#### Somatic Variant Calling

Small variants are called using multiple tools: Strelka2 for single nucleotide variants (SNVs) and small insertions/deletions (INDELs), GATK Mutect2 for SNVs, multinucleotide variants greater than 1bp in length (MNVs) and INDELs, Lancet for SNVs, MNVs, and INDELs, and VarDict Java which calls SNVs, MNVs, INDELs and more. 
Larger copy number variants are also called using multiple tools as well: ControlFreeC, CNVkit and GATK CNV. 

> CNVkit calls are adjusted for purity estimations using THeTa2. 

Author: So purity isn't considered in any of the calculations currently?

> CNVkit calls are adjusted for purity estimations using THeTa2. 

Reviewer: Correct, it is a product of the algorithm

Manta is also used to determine structural variants (SVs) and INDELs. 
All calls are made using GRCh38 references and variants are then annotated using gnomAD and for cancer hot spots. 
Publicly available files are further subjected to “germline masking,” which removes low frequency variants that could be used to identify the sample donor. 
For more details see the GitHub release at [Kid's First Somatic Variant Workflow](https://github.com/kids-first/kf-somatic-workflow) or to run the pipeline see the [CAVATICA App](https://cavatica.sbgenomics.com/public/apps/cavatica/apps-publisher/kfdrc-somatic-variant-workflow). 
Once in the Cavatica workflow page, please click on the "Read All" link to open up the full documentation. 

> Panel data was not reprocessed and the variant files provided by the submitters were merged with our harmonized data files.

Reviewer: ```suggestion
Panel data was not reprocessed and the variant files provided by the submitters were merged with our harmonized data files.
```

TARGET panel data was reprocessed, and DGD was somewhat - will have to get some details on current DGD processing

Author: I had confirmed with Bo that they just took the MAF files for DGD. @zhangb1 is that correct, that you didn't reprocess any DGD files?

Reviewer: DGD processed in hg19, so it is a bit complicated. We had rerun their workflow (i believe) in hg38 but then we missed an artifact filtering step - cc @luederm - can you inform on the current DGD release files what was done prior to your coming aboard?

The majority of Pediatric Molecular Target data is paired-end, but single end methods are provided if you want to apply the pipeline to your own data. 
Sequencing quality is checked using FastQC and NGSCheckMate is used to confirm tumor/normal DNA and normal DNA/ tumor RNA pairs come from the same individual. 
For more details on sample identity confirmation please see the [NGS Checkmate Workflow](https://github.com/d3b-center/OpenPBTA-workflows/blob/master/cwl/ngs_checkmate_wf.cwl). 
Somatic variants are called using Mutect2, Strelka2, Lancet, and VarDict, followed by a 2/4 consensus and hotspot scavenging.
For more detailed alignment and variant calling methods, please refer to the [OpenPBTA manuscript](https://www.biorxiv.org/content/10.1101/2022.09.13.507832v1). 
To run the CAVATICA apps yourself, see the [OpenPBTA workflows repository](https://github.com/d3b-center/OpenPBTA-workflows/tree/master/cwl) and the [CAVATICA App](https://cavatica.sbgenomics.com/public/apps/cavatica/apps-publisher/kfdrc-alignment-workflow). 
Once in the Cavatica workflow page, please click on the "Read All" link to open up the full documentation.

> CHOP P30 targeted sequencing panel data was processed using the GRCh37 reference with CHOP's Division of Genomic Diagnostics published pipeline and then coordinates were lifted over to GRCh38. For more details on the panel design and variant calling pipeline, please see the methods in the [Genome Medicine paper](https://doi.org/10.1186/s13073-019-0644-8).

Author: @jharenza DGD panel processing description, re: https://github.com/PediatricOpenTargets/documentation/pull/8#discussion_r1036579470

Author: Also https://github.com/PediatricOpenTargets/documentation/pull/8#discussion_r1036343020

| Variant ID hg38 | Specific name for the variant in human genome hg38 coordinates; for example chr12_94581668_T_C means that base 94581668 on chromosome 12 is not the reference thymine (T) but mutated to a cytosine (C) |  |
| Protein change | Amino acid change if mutation causes one; for example p.R317G means that the 317th amino acid is changed from arginine (R) to glycine (G) |  |
| PMTL | Whether the gene is a relevant target on the PMTL (Pediatric Molecular Target List) | Binary; either an **R** for relevant target or **NR** for non-relevant target and left **blank** if no data |

> | Dataset | See the Dataset section in this document for more details | **All Cohorts** = all datasets combined except for panel data, **TARGET** = Therapeutically Applicable Research to Generate Effective Treatments, **TARGET panel** = same as TARGET but the cohort sequencing panel data only, **PBTA** = Pediatric Brain Tumor Atlas, **GMFK** = Gabriella Miller Kids First Neuroblastoma |

Reviewer: ```suggestion
| Dataset | See the Dataset section in this document for more details | **All Cohorts** = all datasets combined except for panel data, **TARGET** = Therapeutically Applicable Research to Generate Effective Treatments, **TARGET panel** = same as TARGET but the cohort sequencing panel data only, **PBTA** = Pediatric Brain Tumor Atlas, **GMKF** = Gabriella Miller Kids First Neuroblastoma |
```

#### Copy Number Variants (CNVs)

Multiple callers were used to determine copy number variants (CNVs). 

> A consensus CNV file was constructed, consisting of CNVs that were called by 2 or more copy number callers: ControlFreeC, CNVkit, and GATK CNV. 

Reviewer: I do not believe that our modules used the consensus using GATK CNV for tables, but let's check with @ewafula - we may just update this for v12 (as in, it is not in the release using GATK CNV)

Author: Eric confirmed that this was correct on Slack

> A consensus CNV file was constructed, consisting of CNVs that were called by 2 or more copy number callers: ControlFreeC, CNVkit, and GATK CNV. 

Reviewer: ```suggestion
A consensus CNV file was constructed, consisting of CNVs that were called by 2 or more copy number callers: ControlFreeC, CNVkit, and Manta SV. 
```

> See https://github.com/kids-first/kf-somatic-workflow/blob/master/docs/kfdrc-consensus-calling.md for more detail on how the calls were combined. 

Reviewer: This
```suggestion
See the [copy_number_consensus_call GitHub module] (https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/29c344bc237eb5dd73f65cb7c1a04ae0c37414df/analyses/copy_number_consensus_call) for more detail on how the calls were combined. 
```
that link is SNV only (they did not yet implement CNV consensus for KF)
I added the specific commit for our v11 release here, so it includes manta info

The various nomenclatures used by the different callers are harmonized to standard descriptions of CNVs: deep deletion for when copy number equals zero, loss for when copy number is less than ploidy, neutral if copy number is the same as the genome copy number (cancer genomes may have a ploidy other than two), gain for up to two times ploidy, and amplification for a gain of more than two times ploidy. 
Copy number variants were retained only if they overlapped by at least 1 base pair with a gene’s exon. 
Further annotations, including alternative gene and protein IDs and cancer references, were also added.

Then several variant frequencies were calculated for each of those genes within each cancer group and cohort. 
The frequency in the overall dataset is the percentage of patients that have that a CNV affecting that gene in the given cohort out of all patients in that cohort. 
The frequency in primary or relapse tumors is the percentage of samples that that have that a CNV affecting that gene in the given cohort out of all samples in that cohort. 
Note that the frequencies and counts may not tally as expected for several reasons. 

> First, the total
>  columns use unique patients, while the primary/relapse tumor columns use unique samples. Second, some submitters did not include information about the primary/relapse status of the samples, so those samples are omitted from the primary/relapse counts. 

Reviewer: ```suggestion
First, the total columns use unique patients, while the primary/relapse tumor columns use unique samples. Second, some submitters did not include information about the primary/relapse status of the samples, so those samples are omitted from the primary/relapse counts. 
```

| Variant type | Categorical description of the variant type; cancer genomes may have a ploidy other than diploid which is why the categories are described in terms of the ploidy of the sample | **deep deletion** = 0 copies, **loss** = fewer copies than ploidy, **neutral** = same as ploidy, **gain** = up to 2 times ploidy, **amplification** = more than 2 times ploidy |
| Variant category |  |  |

> | Dataset | See the Dataset section in this document for more details | **All Cohorts** = all datasets combined, **TARGET** = Therapeutically Applicable Research to Generate Effective Treatments, **PBTA** = Pediatric Brain Tumor Atlas, **GMFK** = Gabriella Miller Kids First Neuroblastoma  |

Reviewer: ```suggestion
| Dataset | See the Dataset section in this document for more details | **All Cohorts** = all datasets combined, **TARGET** = Therapeutically Applicable Research to Generate Effective Treatments, **PBTA** = Pediatric Brain Tumor Atlas, **GMKF** = Gabriella Miller Kids First Neuroblastoma  |
```

---

## RNA_Sequencing

### Data Processing

> The RNA-seq Alignment Workflow begins by trimming adapters, only if adapters are provided, using Cutadapt. 

Reviewer: ```suggestion
The RNA-seq Alignment Workflow begins by trimming adapters, only if adapters are provided, using Cutadapt. 
```
I believe Cutadapt will automatically detect the adapters and trim - @zhangb1 can you confirm?

> Sequencing quality is checked using FastQC and tumor/normal pairs are double-checked to confirm they are from the same individual using NGSCheckMate. 

Reviewer: ```suggestion
```
Think we technically use an RNAFastQC (not sure of exact name) but I think it is fine to remove this as it is stated generally above

> For more details on sample identity confirmation please see the [Kid's First NGS Checkmate Workflow](https://github.com/kids-first/ngs_checkmate_wf). 

Reviewer: ```suggestion
```

> Reads were then aligned using STAR in two-pass mode to reference genome GRCh38. While all MTP data is paired-end, methods are provided for single-end alignment if you are interested in processing your data in the same manner. 

Reviewer: ```suggestion
Reads were then aligned using STAR in two-pass mode to reference genome GRCh38. 
While all MTP data is paired-end, methods are provided for single-end alignment if you are interested in processing your data in the same manner. 
```

> Transcripts are quantified using RSEM at both the gene and isoform level  with the GENCODE v27 annotation, except for the GTEx samples which were not re-processed and are annotated using GENCODE v26. 

Reviewer: ```suggestion
Transcripts are quantified using RSEM at both the gene and isoform level  with the GENCODE v27 annotation, except for the GTEx v8 samples which were not re-processed and are annotated using GENCODE v26. 
```

> Fusion calling is done using both Arriba and STAR-Fusion and then filtered for high confidence fusion calls using annoFuse. 

Reviewer: also not urgent, but may be useful to be linking to software pages or papers, but this will all be done in manuscript

> QC metrics for the alignment are summarized using RNA-seQC. 

Reviewer: oh, here it is!

> If you would like to view the code in more detail, please see the GitHub release [Kids First RNA-seq Workflow](https://github.com/kids-first/kf-rnaseq-workflow) and if you would like to run the pipeline, please see the [CAVATICA App](https://cavatica.sbgenomics.com/public/apps/cavatica/apps-publisher/kfdrc-rnaseq-workflow/6). 

Reviewer: ```suggestion
If you would like to view the code in more detail, please see the GitHub release [OpenPBTA RNA-seq Workflow](https://github.com/d3b-center/OpenPBTA-workflows/blob/master/cwl/kfdrc_RNAseq_workflow.cwl) and if you would like to run the pipeline, please see the [CAVATICA App](https://cavatica.sbgenomics.com/public/apps/cavatica/apps-publisher/kfdrc-rnaseq-workflow/6). 
```

Once in the Cavatica workflow page, please click on the "Read All" link to open up the full documentation. 

### RNA Sequencing Data

#### Fusions

> Gene fusions are called solely from RNA sequencing using the programs above. 

Reviewer: This is now not accurate since we added DGD fusion data - those were called using Archer Dx fusion panels, so can you add that as well?

> Fusions are filtered using custom R scripts. 

Reviewer: ```suggestion
```

I don't like using "custom" script language for reproducible work such as this. We can instead refer to our module for fusion filtering and prioritization.

> Fusion calls are retained if they are called by both STAR-Fusion and Arriba and if the fusion was specific and present in 3 or more samples in a single disease. 

Reviewer: This is also not true- 
```suggestion
We use annoFuse to filter out artifactual fusions, those with both gene partners having expression < 1 TPM, or fusions present in normal tissues. 
We annotate genes as transcription factor, tumor suppressor, oncogene, and/or kinase and for kinase genes, identify whether or not the kinase domain is retained in the fusion.
Fusions present in more than 4 broad histologies are also flagged and filtered out, as they may be non-specific.
This final file of putative oncogenic fusions is the data utilized for the fusion frequency tables.
```

See here: https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/fusion_filtering/05-QC_putative_onco_fusion_distribution.Rmd

Fusion panels were not reprocessed and were merged with our harmonized fusion calls.

> Fusions were then annotated with gene and fusion specific information as well as whether they are known cancer genes from OncoKB, TCGA, and COSMIC. 

Reviewer: ```suggestion
```

I added this above and that's not 100% accurate as the data sources were differerent

Once in the Cavatica workflow page, please click on the "Read All" link to open up the full documentation. 

### RNA Sequencing Data

#### Fusions

Gene fusions are called either from NGS RNA sequencing using the methods described above or for the CHOP P30 panel data, using CHOP's Division of Genome Diagnostic's (DGD) Fusion Gene Panel. 
No whole genome or whole exome NGS DNA sequencing is used to call fusions currently.
We use annoFuse to filter out artifactual fusions, those with both gene partners having expression < 1 TPM, or fusions present in normal tissues. 
We annotate genes as transcription factor, tumor suppressor, oncogene, and/or kinase and for kinase genes, identify whether or not the kinase domain is retained in the fusion.
Fusions present in more than 4 broad histologies are also flagged and filtered out, as they may be non-specific.
This final file of putative oncogenic fusions is the data utilized for the fusion frequency tables.
Fusion panels were not reprocessed and were merged with our harmonized fusion calls.
Summary frequencies are calculated using R. 

> See [the fusion filtering documentation](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/fusion_filtering) for specific code and further details. 

Author: Does this updated paragraph address https://github.com/PediatricOpenTargets/documentation/pull/8#discussion_r1036587598 on fusion methods and https://github.com/PediatricOpenTargets/documentation/pull/8#discussion_r1036588027 on custom script language?

Reviewer: great!

---

## RNA_Sequencing

### Data Processing

> The RNA-seq Alignment Workflow begins by trimming adapters, if necessary, using Cutadapt. 

Author: It's correct that Cutadapt can automatically detect adapters. Updated the description to say "if necessary" instead to address https://github.com/PediatricOpenTargets/documentation/pull/8#discussion_r1036585697 which matches the description on CAVATICA https://cavatica.sbgenomics.com/public/apps/cavatica/apps-publisher/kfdrc-rnaseq-workflow/6

| Gene2A Annotation | A limited set of simplified annotations for the second gene on the 5’ side of an intergenic fusion | **CosmicCensus** = fusion is in COSMIC, **Kinase** = one of the fusion genes is a kinase, **Oncogene** = one gene is an known oncogene, **TranscriptionFactor** = one of the fusion genes is a transcription factor, **TumorSuppressorGene** = one gene is a known tumor suppressor, left **blank** if no annotations applyy |
| Gene2B Annotation | A limited set of simplified annotations for the second gene on the 3’ side of an intergenic fusion | **CosmicCensus** = fusion is in COSMIC, **Kinase** = one of the fusion genes is a kinase, **Oncogene** = one gene is an known oncogene, **TranscriptionFactor** = one of the fusion genes is a transcription factor, **TumorSuppressorGene** = one gene is a known tumor suppressor, left **blank** if no annotations apply |
| Gene Ensembl ID | Official gene ID from Ensembl  | |

> | Dataset | See the Dataset section in this document for more details | **All Cohorts** = all datasets combined except for panel data, **TARGET** = Therapeutically Applicable Research to Generate Effective Treatments, **PBTA** = Pediatric Brain Tumor Atlas, **GMFK** = Gabriella Miller Kids First Neuroblastoma, **CHOP P30 panel** |

Reviewer: ```suggestion
| Dataset | See the Dataset section in this document for more details | **All Cohorts** = all datasets combined except for panel data, **TARGET** = Therapeutically Applicable Research to Generate Effective Treatments, **PBTA** = Pediatric Brain Tumor Atlas, **GMKF** = Gabriella Miller Kids First Neuroblastoma, **CHOP P30 panel** |
```
